lib/smart_proxy_dns_powerdns/dns_powerdns_main.rb

```diff
@@ -15,15 +15,23 @@ def initialize(a_server = nil, a_ttl = nil)
     end
 
     def create_a_record(fqdn, ip)
-      if found = dns_find(fqdn)
+      if found = dns_find(fqdn, 'A')
         raise Proxy::Dns::Collision, "#{fqdn} is already in use by #{ip}" unless found == ip
       end
 
       do_create(fqdn, ip, "A")
     end
 
+    def create_aaaa_record(fqdn, ip)
+      if found = dns_find(fqdn, 'AAAA')
```

Comment:
see my comment in your PR in smart-proxy: this isn't going to work for shortened ipv6 addresses.